### PR TITLE
feat: add grpc client timeout

### PR DESCRIPTION
**Pull request by peng09**

### CODE REVIEW

File: deploy/kubernetes_operator/pkg/operator/event_handler.go [hunk 1]
```diff
@@ -164,11 +185,19 @@ func (handler *appEventHandler) Finish(ctx context.Context, app *v1alpha1.FLApp)
...
+			ctx, cancel := newContextWithHeaders(peerSpec.ExtraHeaders)
+			defer cancel()
+			response, err := client.Finish(ctx, request)
+			if err != nil || response.Code != int32(codes.OK) {
+				return fmt.Errorf("Finish failed name = %v, role = %v, err = %v", name, role, err), ""
+			}
+			return nil, response.ErrorMessage
+		}()
+		if err != nil {
+			return err
```
Comment (DeliangFan) [inline deploy/kubernetes_operator/pkg/operator/event_handler.go:198]: what about abstract a new func to reduce the duplicated processes.

File: deploy/kubernetes_operator/pkg/operator/event_handler.go [hunk 2]
```diff
@@ -33,6 +34,8 @@ import (
 	pb "github.com/bytedance/fedlearner/deploy/kubernetes_operator/proto"
 )
 
+const clientTimeout = time.Second * 1
```
Comment (DeliangFan) [inline deploy/kubernetes_operator/pkg/operator/event_handler.go:37]: May be one second is little short for public network connection some times. 3-5 seconds may be better.

Comment (peng09) [reply deploy/kubernetes_operator/pkg/operator/event_handler.go]: Good point, fixed

**Comment by DeliangFan** [comment]
lgtm

**Pull request merged**